lib/PSR/Storage/SQLite.hs

```diff
@@ -90,6 +105,17 @@ mkStorage conn = do
                 paramsWithQueries =
                     catMaybes
                         [ fmap
+                            ( mkNamedParam
+                                " ((CASE \
+                                \ WHEN :event_type = 'execution' THEN e.block_id \
+                                \ WHEN :event_type = 'cancellation' THEN c.block_id \
+                                \ WHEN :event_type = 'selection' THEN s.block_id \
+                                \ END) IS NOT NULL) "
+                                ":event_type"
+                                . toField
+                            )
+                            _eventFilterParam_type
+                        , fmap
```

Comment:
I was going to suggest this style, it makes it clearer what the subject of the expression is:

```suggestion
                        [_eventFilterParam_type <&>
                            mkNamedParam
                                " ((CASE \
                                \ WHEN :event_type = 'execution' THEN e.block_id \
                                \ WHEN :event_type = 'cancellation' THEN c.block_id \
                                \ WHEN :event_type = 'selection' THEN s.block_id \
                                \ END) IS NOT NULL) "
                                ":event_type"                           
                        , fmap
```

Also the `.  toField` isn't needed, the `:=` operator takes care of that already.